src/main/java/org/apache/bcel/verifier/statics/Pass2Verifier.java

```diff
@@ -203,7 +203,7 @@ private void every_class_has_an_accessible_superclass() {
             supidx = jc.getSuperclassNameIndex();
 
             if (supidx == 0) {
-                if (jc != Repository.lookupClass(Type.OBJECT.getClassName())) {
+                if (!jc.equals(Repository.lookupClass(Type.OBJECT.getClassName()))) {
```

Comment:
Ordinarily I prefer to avoid code changes to the upstream repository.  In this case, I think a better solution would be to annotate the return type of `Repository.lookupClass` as `@Interned`.